openedx_events/learning/data.py

```diff
@@ -483,3 +483,25 @@ class UserCourseData:
 
     user = attr.ib(type=UserData)
     course = attr.ib(type=CourseData)
+
+
+@attr.s(frozen=True)
+class BadgeData:
+    """
+    Attributes defined for the Open edX badge data object.
+
+    Arguments:
+        uuid (str): the UUID of the badge
+        user (UserData): user associated with the badge
+        badge_type (str): type of badge
+        name (str): badge name
+        description (str): badge description
+        image_url (str): badge image url
+    """
+
+    uuid = attr.ib(type=str)
+    user = attr.ib(type=UserData)
```

Comment:
NOTE: to be able to put all user attributes we'll have to store user PII data in Credentials DB (currently, only `username` is stored in `UserCredential`).